src/trading/Order.sol

```diff
@@ -129,46 +129,46 @@ library Order {
                 }
             }
         }
         // If not able to cover entire order with shares alone, then cover remaining with tokens
         if (_attosharesToCover > 0) {
-            _orderData.fxpMoneyEscrowed = SafeMathUint256.fxpMul(_attosharesToCover, uint256(_orderData.fxpPrice), 1 ether);
-            require(_orderData.market.getDenominationToken().transferFrom(_orderData.maker, _orderData.market, _orderData.fxpMoneyEscrowed));
+            _orderData.moneyEscrowed = _attosharesToCover.mul(_orderData.price);
+            require(_orderData.market.getDenominationToken().transferFrom(_orderData.maker, _orderData.market, _orderData.moneyEscrowed));
         }
 
         return true;
     }
 
     function escrowFundsForAsk(Order.Data _orderData) private returns (bool) {
-        uint256 _orderValueInAttotokens = SafeMathUint256.fxpMul(_orderData.fxpAmount, uint256(_orderData.fxpPrice), 1 ether);
+        uint256 _orderValueInAttotokens = _orderData.amount.mul(_orderData.price);
         require(_orderValueInAttotokens >= MIN_ORDER_VALUE);
 
-        require(_orderData.fxpMoneyEscrowed == 0);
-        require(_orderData.fxpSharesEscrowed == 0);
+        require(_orderData.moneyEscrowed == 0);
+        require(_orderData.sharesEscrowed == 0);
         IShareToken _shareToken = _orderData.market.getShareToken(_orderData.outcome);
-        uint256 _attosharesToCover = _orderData.fxpAmount;
+        uint256 _attosharesToCover = _orderData.amount;
 
         // Figure out how many shares of the outcome the maker has
         uint256 _attosharesHeld = _shareToken.balanceOf(_orderData.maker);
 
         // Take shares in escrow if user has shares
         if (_attosharesHeld > 0) {
-            _orderData.fxpSharesEscrowed = SafeMathUint256.min(_attosharesHeld, _attosharesToCover);
-            _attosharesToCover -= _orderData.fxpSharesEscrowed;
-            _shareToken.transferFrom(_orderData.maker, _orderData.market, _orderData.fxpSharesEscrowed);
+            _orderData.sharesEscrowed = SafeMathUint256.min(_attosharesHeld, _attosharesToCover);
+            _attosharesToCover -= _orderData.sharesEscrowed;
+            _shareToken.transferFrom(_orderData.maker, _orderData.market, _orderData.sharesEscrowed);
         }
 
         // If not able to cover entire order with shares alone, then cover remaining with tokens
         if (_attosharesToCover > 0) {
-            _orderData.fxpMoneyEscrowed = SafeMathUint256.fxpMul(_attosharesToCover, SafeMathUint256.sub(_orderData.market.getCompleteSetCostInAttotokens(), uint256(_orderData.fxpPrice)), 1 ether);
-            require(_orderData.market.getDenominationToken().transferFrom(_orderData.maker, _orderData.market, _orderData.fxpMoneyEscrowed));
+            _orderData.moneyEscrowed = _attosharesToCover.mul(SafeMathUint256.sub(_orderData.market.getMarketDenominator(), _orderData.price));
```

Comment:
Same as comment above about extension calling convention and chaining over nesting.